# Upstream status: in 2.6.18-rc2

This fixes a fatal typo which crept in the rewritten interrupt handler.

Signed-off-by: Thiemo Seufer <ths@networkno.de>

--- a/arch/mips/sibyte/bcm1480/irq.c
+++ b/arch/mips/sibyte/bcm1480/irq.c
@@ -533,7 +533,7 @@ #endif
 		mask_l = __raw_readq(
 			IOADDR(base + R_BCM1480_IMR_INTERRUPT_STATUS_BASE_L));
 
-		if (!mask_h) {
+		if (mask_h) {
 			if (mask_h ^ 1)
 				do_IRQ(63 - dclz(mask_h), regs);
 			else



